glab panics on `glab issue list --mine` if not authenticated

**Description**
I tried running `glab issue list --mine` in a repo with an unauthenticated hostname. Since the `--mine` flag looks for the user ID of the current authenticated user, it panics without giving any useful information to the user.

**Expected Behavior**
The user should be notified that they need to authenticate `glab` for the hostname or return `unauthorized error`

**Actual Behavior**
```sh
→ glab issue list --mine -R gitlab-com/www-gitlab-com
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xc4bcc1]

goroutine 1 [running]:
github.com/profclems/glab/commands/issue/list.NewCmdList.func1(0xc0006f2840, 0xc000749dd0, 0x0, 0x3, 0x0, 0x0)
        github.com/profclems/glab/commands/issue/list/issue_list.go:85 +0x941
github.com/spf13/cobra.(*Command).execute(0xc0006f2840, 0xc000749da0, 0x3, 0x3, 0xc0006f2840, 0xc000749da0)
        github.com/spf13/cobra@v1.1.1/command.go:850 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002acdc0, 0xc00000e018, 0xc4, 0x5)
        github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
main.main()
        github.com/profclems/glab/cmd/glab/main.go:116 +0x793

```

**Possible Fix**
<!--- Not obligatory, but suggest a fix or reason for the bug -->
The error returned by the user service is explicitly ignored resulting in the panic.
https://github.com/profclems/glab/blob/96bfb8ed06fdbf8f6d437716e5723aba41b7d470/commands/issue/list/issue_list.go#L83-L87

Could be resolved as this:
```go
u, err := api.CurrentUser(nil)
if err != nil {
    return err
}
```

**Steps to Reproduce**
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1. Authenticate a hostname with a wrong token
2. Run `glab issue list --mine`
3. `glab` panics


**Your Environment**
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version used: v1.12.1
* Operating System and version: Arch Linux x86_64 (5.9.12-arch1-1)

--- a/commands/issue/list/issue_list.go
+++ b/commands/issue/list/issue_list.go
@@ -81,7 +81,10 @@ func NewCmdList(f *cmdutils.Factory) *cobra.Command {
 			}
 
 			if lb, _ := cmd.Flags().GetBool("mine"); lb {
-				u, _ := api.CurrentUser(nil)
+				u, err := api.CurrentUser(nil)
+				if err != nil {
+					return err
+				}
 				opts.AssigneeUsername = gitlab.String(u.Username)
 				listType = "search"
 			}

